### ISSUE TITLE
Intellisense problem with FSX interactive

### ISSUE THREAD

**Issue by parmigianig** [ionide/ionide-vscode-fsharp#362]
Hi,

intellisense is not working as expected with type providers when i have referenced them within a packet folder. the same is working in visual studio with no solution at all, just file .fsx.

for example  

http://fsprojects.github.io/FSharp.Management/FileSystemProvider.html

with this type provider, in visual studio i am able to see all list of files and directories,
wheras i'm not in vscode.

i tryed also with Atom editor, getting the same result. is it just me?

thank you.

here's my test script, replace any file reference with whatever you have on your filesystem.

```fs

open System
open System.IO
 
Environment.CurrentDirectory <- __SOURCE_DIRECTORY__
 
if not (File.Exists "paket.exe") then
    let url = "http://fsprojects.github.io/Paket/stable"
    use wc = new Net.WebClient() 
    let tmp = Path.GetTempFileName() 
    let stable = wc.DownloadString(url) 
    wc.DownloadFile(stable, tmp)
    File.Move(tmp,Path.GetFileName stable)

#r "paket.exe"
open Paket

// locate the paket.dependencies file
let dependencies = Dependencies.Locate(__SOURCE_DIRECTORY__)
dependencies.Add "FSharp.Data"
dependencies.Add "FSharp.Management"
dependencies.Add "System.Xml.Linq"

UpdateProcess.Update(dependencies.DependenciesFile, UpdaterOptions.Default);

[<Literal>]
let CurrentDirectory = __SOURCE_DIRECTORY__

// reference the type provider dll
#r @".\packages\FSharp.Management\lib\net40\FSharp.Management.dll"
#r @".\packages\FSharp.Data\lib\net40\FSharp.Data.dll"
#r @".\packages\System.Xml.Linq\lib\net20\System.Xml.Linq.dll"

open FSharp.Management
open FSharp.Data
open System.Xml.Linq

// Let the type provider do it's work
type Local = FileSystem< CurrentDirectory, watch = true >
type CsvReader = CsvProvider< Local.test.``SacramentocrimeJanuary2006.csv``>
type XmlReader = XmlProvider< Local.test.``sample1.xml`` >


let data = 
    CsvReader.Load(Local.test.``SacramentocrimeJanuary2006.csv``).Rows
    |> Seq.filter (fun r -> r.Cdatetime > DateTime.Parse("28/01/2006 00:00:00")) 
in
for r in data do
    printfn "%A" (r.Address, r.Cdatetime, r.Crimedescr, r.District)
```

**Comment by jindraivanek** [comment]
I was able to track down this bug. FSAC throw an exception in `DescriptionText` call [here](https://github.com/ionide/FsAutoComplete/blob/master/src/FsAutoComplete.Core/Commands.fs#L193).

If I remove `DescriptionText` calls, Intellisense works.

Stack trace:
```
Unhandled error - One or more errors occurred. 
   at System.Threading.Tasks.Task.ThrowIfExceptional (System.Boolean includeTaskCanceledExceptions) [0x00014] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Threading.Tasks.Task.Wait (System.Int32 millisecondsTimeout, System.Threading.CancellationToken cancellationToken) [0x00052] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Threading.Tasks.Task.Wait (System.Int32 millisecondsTimeout) [0x00000] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at Microsoft.FSharp.Compiler.SourceCodeServices.FSharpDeclarationListItem.get_StructuredDescriptionText () [0x00069] in <58c02e00ddab8ea7a7450383002ec058>:0 
  at Microsoft.FSharp.Compiler.SourceCodeServices.FSharpDeclarationListItem.get_DescriptionText () [0x00001] in <58c02e00ddab8ea7a7450383002ec058>:0 
  at <StartupCode$FsAutoComplete-Core>.$Commands+Completion@181-1.Invoke (Microsoft.FSharp.Core.FSharpOption`1[T] _arg7) [0x00089] in /pub/dev/github/ionide/FsAutoComplete/src/FsAutoComplete.Core/Commands.fs:193 
  at Microsoft.FSharp.Control.AsyncBuilderImpl+args@825-1[a,b].Invoke (a a) [0x00016] in <5893d081904cf4daa745038381d09358>:0 
[E] 2017-03-24T16:20:48.7018330Z: request failed [Suave.Web.defaultErrorHandler] exn:
System.Runtime.Serialization.SerializationException: Type 'System.Exception' with data contract name 'Exception:http://schemas.datacontract.org/2004/07/System' is not expected. Add any types not known statically to the list of known types - for example, by using the KnownTypeAttribute attribute or by adding them to the list of known types passed to DataContractSerializer.
  at System.Runtime.Serialization.XmlObjectSerializerWriteContext.SerializeAndVerifyType (System.Runtime.Serialization.DataContract dataContract, System.Runtime.Serialization.XmlWriterDelegator xmlWriter, System.Object obj, System.Boolean verifyKnownType, System.RuntimeTypeHandle declaredTypeHandle, System.Type declaredType) [0x00079] in <51bb353c45df455c9d11165e7e32cf55>:0 
  at System.Runtime.Serialization.Json.XmlObjectSerializerWriteContextComplexJson.SerializeWithXsiType (System.Runtime.Serialization.XmlWriterDelegator xmlWriter, System.Object obj, System.RuntimeTypeHandle objectTypeHandle, System.Type objectType, System.Int32 declaredTypeID, System.RuntimeTypeHandle declaredTypeHandle, System.Type declaredType) [0x000a7] in <51bb353c45df455c9d11165e7e32cf55>:0 
  at System.Runtime.Serialization.XmlObjectSerializerWriteContext.InternalSerialize (System.Runtime.Serialization.XmlWriterDelegator xmlWriter, System.Object obj, System.Boolean isDeclaredType, System.Boolean writeXsiType, System.Int32 declaredTypeID, System.RuntimeTypeHandle declaredTypeHandle) [0x0009f] in <51bb353c45df455c9d11165e7e32cf55>:0 
  at System.Runtime.Serialization.XmlObjectSerializerWriteContextComplex.InternalSerialize (System.Runtime.Serialization.XmlWriterDelegator xmlWriter, System.Object obj, System.Boolean isDeclaredType, System.Boolean writeXsiType, System.Int32 declaredTypeID, System.RuntimeTypeHandle declaredTypeHandle) [0x0000b] in <51bb353c45df455c9d11165e7e32cf55>:0 
  at System.Runtime.Serialization.XmlObjectSerializerWriteContext.InternalSerializeReference (System.Runtime.Serialization.XmlWriterDelegator xmlWriter, System.Object obj, System.Boolean isDeclaredType, System.Boolean writeXsiType, System.Int32 declaredTypeID, System.RuntimeTypeHandle declaredTypeHandle) [0x0000e] in <51bb353c45df455c9d11165e7e32cf55>:0 
  at System.Runtime.Serialization.XmlObjectSerializerWriteContext.WriteSerializationInfo (System.Runtime.Serialization.XmlWriterDelegator xmlWriter, System.Type objType, System.Runtime.Serialization.SerializationInfo serInfo) [0x000fc] in <51bb353c45df455c9d11165e7e32cf55>:0 
  at System.Runtime.Serialization.Json.XmlObjectSerializerWriteContextComplexJson.WriteJsonISerializable (System.Runtime.Serialization.XmlWriterDelegator xmlWriter, System.Runtime.Serialization.ISerializable obj) [0x00064] in <51bb353c45df455c9d11165e7e32cf55>:0 
  at System.Runtime.Serialization.Json.JsonFormatWriterInterpreter.WriteClass (System.Runtime.Serialization.ClassDataContract classContract) [0x0001e] in <51bb353c45df455c9d11165e7e32cf55>:0 
  at System.Runtime.Serialization.Json.JsonFormatWriterInterpreter.WriteToJson (System.Runtime.Serialization.XmlWriterDelegator xmlWriter, System.Object obj, System.Runtime.Serialization.Json.XmlObjectSerializerWriteContextComplexJson context, System.Runtime.Serialization.ClassDataContract dataContract, System.Xml.XmlDictionaryString[] memberNames) [0x00057] in <51bb353c45df455c9d11165e7e32cf55>:0 
  at System.Runtime.Serialization.Json.JsonFormatWriterGenerator+CriticalHelper+<GenerateClassWriter>c__AnonStorey0.<>m__0 (System.Runtime.Serialization.XmlWriterDelegator xmlWriter, System.Object obj, System.Runtime.Serialization.Json.XmlObjectSerializerWriteContextComplexJson context, System.Runtime.Serialization.ClassDataContract dataContract, System.Xml.XmlDictionaryString[] memberNames) [0x0000b] in <51bb353c45df455c9d11165e7e32cf55>:0 
  at System.Runtime.Serialization.Json.JsonClassDataContract.WriteJsonValueCore (System.Runtime.Serialization.XmlWriterDelegator jsonWriter, System.Object obj, System.Runtime.Serialization.Json.XmlObjectSerializerWriteContextComplexJson context, System.RuntimeTypeHandle declaredTypeHandle) [0x00027] in <51bb353c45df455c9d11165e7e32cf55>:0 
  at System.Runtime.Serialization.Json.JsonDataContract.WriteJsonValue (System.Runtime.Serialization.XmlWriterDelegator jsonWriter, System.Object obj, System.Runtime.Serialization.Json.XmlObjectSerializerWriteContextComplexJson context, System.RuntimeTypeHandle declaredTypeHandle) [0x00007] in <51bb353c45df455c9d11165e7e32cf55>:0 
  at System.Runtime.Serialization.Json.DataContractJsonSerializer.WriteJsonValue (System.Runtime.Serialization.Json.JsonDataContract contract, System.Runtime.Serialization.XmlWriterDelegator writer, System.Object graph, System.Runtime.Serialization.Json.XmlObjectSerializerWriteContextComplexJson context, System.RuntimeTypeHandle declaredTypeHandle) [0x00000] in <51bb353c45df455c9d11165e7e32cf55>:0 
  at System.Runtime.Serialization.Json.XmlObjectSerializerWriteContextComplexJson.WriteDataContractValue (System.Runtime.Serialization.DataContract dataContract, System.Runtime.Serialization.XmlWriterDelegator xmlWriter, System.Object obj, System.RuntimeTypeHandle declaredTypeHandle) [0x0003d] in <51bb353c45df455c9d11165e7e32cf55>:0 
  at System.Runtime.Serialization.XmlObjectSerializerWriteContext.SerializeWithoutXsiType (System.Runtime.Serialization.DataContract dataContract, System.Runtime.Serialization.XmlWriterDelegator xmlWriter, System.Object obj, System.RuntimeTypeHandle declaredTypeHandle) [0x00046] in <51bb353c45df455c9d11165e7e32cf55>:0 
  at System.Runtime.Serialization.Json.DataContractJsonSerializer.InternalWriteObjectContent (System.Runtime.Serialization.XmlWriterDelegator writer, System.Object graph) [0x000b5] in <51bb353c45df455c9d11165e7e32cf55>:0 
  at System.Runtime.Serialization.Json.DataContractJsonSerializer.InternalWriteObject (System.Runtime.Serialization.XmlWriterDelegator writer, System.Object graph) [0x00008] in <51bb353c45df455c9d11165e7e32cf55>:0 
  at System.Runtime.Serialization.XmlObjectSerializer.InternalWriteObject (System.Runtime.Serialization.XmlWriterDelegator writer, System.Object graph, System.Runtime.Serialization.DataContractResolver dataContractResolver) [0x00000] in <51bb353c45df455c9d11165e7e32cf55>:0 
  at System.Runtime.Serialization.XmlObjectSerializer.WriteObjectHandleExceptions (System.Runtime.Serialization.XmlWriterDelegator writer, System.Object graph, System.Runtime.Serialization.DataContractResolver dataContractResolver) [0x00079] in <51bb353c45df455c9d11165e7e32cf55>:0 
  at System.Runtime.Serialization.XmlObjectSerializer.WriteObjectHandleExceptions (System.Runtime.Serialization.XmlWriterDelegator writer, System.Object graph) [0x00000] in <51bb353c45df455c9d11165e7e32cf55>:0 
  at System.Runtime.Serialization.Json.DataContractJsonSerializer.WriteObject (System.Xml.XmlDictionaryWriter writer, System.Object graph) [0x0000d] in <51bb353c45df455c9d11165e7e32cf55>:0 
  at System.Runtime.Serialization.Json.DataContractJsonSerializer.WriteObject (System.IO.Stream stream, System.Object graph) [0x00018] in <51bb353c45df455c9d11165e7e32cf55>:0 
  at Suave.Json.toJson[T] (T o) [0x0001c] in <5768094c6d09e09ca74503834c096857>:0 
  at FsAutoComplete.Suave+handler@61-2[b].Invoke (Microsoft.FSharp.Core.FSharpChoice`2[T1,T2] _arg1) [0x0008a] in /pub/dev/github/ionide/FsAutoComplete/src/FsAutoComplete.Suave/FsAutoComplete.Suave.fs:67 
  at Microsoft.FSharp.Control.AsyncBuilderImpl+args@825-1[a,b].Invoke (a a) [0x00016] in <5893d081904cf4daa745038381d09358>:0
```

**Comment by jindraivanek** [comment]
Stack trace from FCS:
```
System.Exception: ILTypeInfo.FromType
  at Microsoft.FSharp.Compiler.Infos+ILTypeInfo.FromType (Microsoft.FSharp.Compiler.TcGlobals+TcGlobals g, Microsoft.FSharp.Compiler.Tast+TType ty) [0x00059] in <58d5680c42303effa74503830c68d558>:0 
  at Microsoft.FSharp.Compiler.SourceCodeServices.ItemDescriptionsImpl.metaInfoOfEntityRef (Microsoft.FSharp.Compiler.InfoReader+InfoReader infoReader, Microsoft.FSharp.Compiler.Range+range m, Microsoft.FSharp.Compiler.Tast+EntityRef tcref) [0x0004a] in <58d5680c42303effa74503830c68d558>:0 
  at Microsoft.FSharp.Compiler.SourceCodeServices.ItemDescriptionsImpl.GetXmlDocSigOfILFieldInfo (Microsoft.FSharp.Compiler.InfoReader+InfoReader infoReader, Microsoft.FSharp.Compiler.Range+range m, Microsoft.FSharp.Compiler.Infos+ILFieldInfo finfo) [0x00014] in <58d5680c42303effa74503830c68d558>:0 
  at Microsoft.FSharp.Compiler.SourceCodeServices.ItemDescriptionsImpl.GetXmlDocHelpSigOfItemForLookup (Microsoft.FSharp.Compiler.InfoReader+InfoReader infoReader, Microsoft.FSharp.Compiler.Range+range m, Microsoft.FSharp.Compiler.NameResolution+Item d) [0x00112] in <58d5680c42303effa74503830c68d558>:0 
  at Microsoft.FSharp.Compiler.SourceCodeServices.ItemDescriptionsImpl.GetXmlCommentForItemAux (Microsoft.FSharp.Core.FSharpOption`1[T] xmlDoc, Microsoft.FSharp.Compiler.InfoReader+InfoReader infoReader, Microsoft.FSharp.Compiler.Range+range m, Microsoft.FSharp.Compiler.NameResolution+Item d) [0x00059] in <58d5680c42303effa74503830c68d558>:0 
  at Microsoft.FSharp.Compiler.SourceCodeServices.ItemDescriptionsImpl.GetXmlCommentForItem (Microsoft.FSharp.Compiler.InfoReader+InfoReader infoReader, Microsoft.FSharp.Compiler.Range+range m, Microsoft.FSharp.Compiler.NameResolution+Item d) [0x0006f] in <58d5680c42303effa74503830c68d558>:0 
  at Microsoft.FSharp.Compiler.SourceCodeServices.ItemDescriptionsImpl.FormatItemDescriptionToToolTipElement (System.Boolean isDecl, Microsoft.FSharp.Compiler.InfoReader+InfoReader infoReader, Microsoft.FSharp.Compiler.Range+range m, Microsoft.FSharp.Compiler.Tastops+DisplayEnv denv, Microsoft.FSharp.Compiler.NameResolution+Item d) [0x00017] in <58d5680c42303effa74503830c68d558>:0 
  at Microsoft.FSharp.Compiler.SourceCodeServices.ItemDescriptionsImpl+FormatStructuredDescriptionOfItem@1195.Invoke (Microsoft.FSharp.Core.Unit unitVar0) [0x00001] in <58d5680c42303effa74503830c68d558>:0 
  at Microsoft.FSharp.Compiler.ErrorScope.Protect[a] (Microsoft.FSharp.Compiler.Range+range m, Microsoft.FSharp.Core.FSharpFunc`2[T,TResult] f, Microsoft.FSharp.Core.FSharpFunc`2[T,TResult] err) [0x00008] in <58d5680c42303effa74503830c68d558>:0 
  at Microsoft.FSharp.Compiler.ErrorLogger+ErrorLoggerExtensions.ReraiseIfWatsonable (System.Exception exn) [0x00040] in <58d5680c42303effa74503830c68d558>:0 
  at Microsoft.FSharp.Compiler.ErrorLogger+ErrorLoggerExtensions.ErrorLogger.ErrorRecovery (Microsoft.FSharp.Compiler.ErrorLogger+ErrorLogger x, System.Exception exn, Microsoft.FSharp.Compiler.Range+range m) [0x00087] in <58d5680c42303effa74503830c68d558>:0
```

**Comment by Krzysztof-Cieslak** [comment]
CC: @dsyme

Any idea why `DescriptionText` may fail like that?

**Comment by jindraivanek** [comment]
This [fix](https://github.com/jindraivanek/FSharp.Compiler.Service/commit/67bd819e50bb8deeef49ab3ae08612de69450ad9) in FCS works for me, although I have no idea if it is the right thing to do :)

**Comment by dsyme** [comment]
@Krzysztof-Cieslak @jindraivanek  I don't understand why the fix works, the call to mateinfoOfTyconRef is protected by an isILAppTy whicch should be sufficient

Still, perhaps there is a case I'm missing.  Could you add an FCS issue with repro steps and submit a PR for the fix?  If yo could isolate out a test case that doesn't involve Ionide or VSCode (just using the FCS API) that would be hugely appreciated.  There is something we need to get to the bottom of here.

**Comment by jindraivanek** [comment]
Issue with repro: https://github.com/fsharp/FSharp.Compiler.Service/issues/734
PR: https://github.com/fsharp/FSharp.Compiler.Service/pull/735

**Comment by jindraivanek** [comment]
Fix is now merged into FCS.

**Comment by Krzysztof-Cieslak** [comment]
Released in `2.25.3`
